src/h.js

```diff
@@ -36,7 +34,7 @@ export function h(nodeName, attributes) {
 			if (simple && lastSimple) {
 				children[children.length-1] += child;
 			}
-			else if (children===EMPTY_CHILDREN) {
+			else if (children===[]) {
```

Comment:
This one actually breaks `shouldComponentUpdate` for leaf nodes, since `props.children` will be different every time, even if it's empty.